crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
`CRYPTO_set_ex_data(to, mx - 1, CRYPTO_get_ex_data(to, mx -1))` would likely be safe, if the index doesn't exist, then CRYPTO_get_ex_data() return NULL, and then the stack would be increased by CRYPTO_set_ex_data()

> but if this succeed all array values are lost anyway.

Can you clarify? The general mechanism during an object dup is to new the ex_data, then dup the ex_data, which may just change the the contents of the ex_data and not the pointer (and subsequently update the **from_d** pointer). The exdatatest.c file does not validate this behavior, as it does not actually set the ex_data value. But I could update that to demonstrate the error.